internal/controller/storageclient_controller.go

```diff
@@ -428,6 +479,100 @@ func (r *storageClientReconcile) reconcilePhases() error {
 			if err != nil {
 				return fmt.Errorf("failed to create remote noobaa: %v", err)
 			}
+		case "StorageClass":
+			storageClass := &storagev1.StorageClass{}
+			storageClass.Name = eResource.Name
+			if err := utils.CreateOrReplace(r.ctx, r.Client, storageClass, func() error {
+				if err := json.Unmarshal(eResource.Data, &storageClass); err != nil {
+					return fmt.Errorf("failed to unmarshal storageclass configuration response: %v", err)
+				}
+				// ensure name even if it was overwritten by unmarshal of whole resource
+				storageClass.Name = eResource.Name
+				removeStorageClaimAsOwner(storageClass)
+				if err := r.own(storageClass); err != nil {
+					return fmt.Errorf("failed to own Storage Class resource: %v", err)
+				}
+				utils.AddLabels(storageClass, eResource.Labels)
+				utils.AddAnnotations(storageClass, eResource.Annotations)
+				return nil
+			}); err != nil {
+				return fmt.Errorf("failed to create or update StorageClass: %s", err)
```

Comment:
The majority of the steps in these new cases (and most of the old cases) are identical and can work on a client.Object interface. Can we please use a util function for all of these steps? (everything after the Unmarshal?

Reply:
done.